README.md
# opvieira.github.com


php/news.txt
PHP                                                                        NEWS
|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
06 Aug 2015, PHP 5.6.12

- Core:
  . Fixed bug #70012 (Exception lost with nested finally block). (Laruence)
  . Fixed bug #70002 (TS issues with temporary dir handling). (Anatol)
  . Fixed bug #69793 (Remotely triggerable stack exhaustion via recursive
    method calls). (Stas)
  . Fixed bug #69892 (Different arrays compare indentical due to integer key
    truncation). (Nikita)
  . Fixed bug #70121 (unserialize() could lead to unexpected methods execution
    / NULL pointer deref). (Stas)

- CLI server:
  . Fixed bug #69655 (php -S changes MKCALENDAR request method to MKCOL). (cmb)
  . Fixed bug #64878 (304 responses return Content-Type header). (cmb)
  
- GD:
  . Fixed bug #53156 (imagerectangle problem with point ordering). (cmb)
  . Fixed bug #66387 (Stack overflow with imagefilltoborder). (cmb)
  . Fixed bug #70102 (imagecreatefromwebm() shifts colors). (cmb)
  . Fixed bug #66590 (imagewebp() doesn't pad to even length). (cmb)
  . Fixed bug #66882 (imagerotate by -90 degrees truncates image by 1px). (cmb)
  . Fixed bug #70064 (imagescale(..., IMG_BICUBIC) leaks memory). (cmb)
  . Fixed bug #69024 (imagescale segfault with palette based image). (cmb)
  . Fixed bug #53154 (Zero-height rectangle has whiskers). (cmb)
  . Fixed bug #67447 (imagecrop() add a black line when cropping). (cmb)
  . Fixed bug #68714 (copy 'n paste error). (cmb)
  . Fixed bug #66339 (PHP segfaults in imagexbm). (cmb)
  . Fixed bug #70047 (gd_info() doesn't report WebP support). (cmb)

- ODBC:
  . Fixed bug #69975 (PHP segfaults when accessing nvarchar(max) defined
    columns). (cmb)

- OpenSSL:
  . Fixed bug #69882 (OpenSSL error “key values mismatch” after 
    openssl_pkcs12_read with extra cert) (Tomasz Sawicki)
  . Fixed bug #70014 (openssl_random_pseudo_bytes() is not cryptographically
    secure). (Stas)

- Phar:
  . Improved fix for bug #69441. (Anatol Belski)
  . Fixed bug #70019 (Files extracted from archive may be placed outside of
    destination directory). (Anatol Belski)

- SOAP:
  . Fixed bug #70081 (SoapClient info leak / null pointer dereference via
    multiple type confusions). (Stas)

- SPL:
  . Fixed bug #70068 (Dangling pointer in the unserialization of ArrayObject
    items). (sean.heelan)
  . Fixed bug #70166 (Use After Free Vulnerability in unserialize() with
    SPLArrayObject). (taoguangchen at icloud dot com)
  . Fixed bug #70168 (Use After Free Vulnerability in unserialize() with
    SplObjectStorage). (taoguangchen at icloud dot com)
  . Fixed bug #70169 (Use After Free Vulnerability in unserialize() with
    SplDoublyLinkedList). (taoguangchen at icloud dot com)

- Standard:
  . Fixed bug #70096 (Repeated iptcembed() adds superfluous FF bytes). (cmb)

09 Jul 2015, PHP 5.6.11

- Core:
  . Fixed bug #69768 (escapeshell*() doesn't cater to !). (cmb)
  . Fixed bug #69703 (Use __builtin_clzl on PowerPC).
    (dja at axtens dot net, Kalle)
  . Fixed bug #69732 (can induce segmentation fault with basic php code).
    (Dmitry)
  . Fixed bug #69642 (Windows 10 reported as Windows 8).
    (Christian Wenz, Anatol Belski)
  . Fixed bug #69551 (parse_ini_file() and parse_ini_string() segmentation
    fault). (Christoph M. Becker)
  . Fixed bug #69781 (phpinfo() reports Professional Editions of Windows
    7/8/8.1/10 as "Business"). (Christian Wenz)
  . Fixed bug #69740 (finally in generator (yield) swallows exception in
    iteration). (Nikita)
  . Fixed bug #69835 (phpinfo() does not report many Windows SKUs).
    (Christian Wenz)
  . Fixed bug #69892 (Different arrays compare indentical due to integer key
    truncation). (Nikita)
  . Fixed bug #69874 (Can't set empty additional_headers for mail()), regression
    from fix to bug #68776. (Yasuo)

- GD:
  . Fixed bug #61221 (imagegammacorrect function loses alpha channel). (cmb)

- GMP:
  . Fixed bug #69803 (gmp_random_range() modifies second parameter if GMP
    number). (Nikita)

- Mysqlnd:
  . Fixed bug #69669 (mysqlnd is vulnerable to BACKRONYM) (CVE-2015-3152).
    (Andrey)
    
- PCRE:
  . Fixed Bug #53823 (preg_replace: * qualifier on unicode replace garbles the
    string). (cmb)
  . Fixed bug #69864 (Segfault in preg_replace_callback) (cmb, ab)

- PDO_pgsql:
  . Fixed bug #69752 (PDOStatement::execute() leaks memory with DML
    Statements when closeCuror() is u). (Philip Hofstetter)
  . Fixed bug #69362 (PDO-pgsql fails to connect if password contains a
    leading single quote). (Matteo)
  . Fixed bug #69344 (PDO PgSQL Incorrect binding numeric array with gaps).
    (Matteo)

- SimpleXML:
  . Refactored the fix for bug #66084 (simplexml_load_string() mangles empty
    node name). (Christoph Michael Becker)

- SPL:
  . Fixed bug #69737 (Segfault when SplMinHeap::compare produces fatal error).
    (Stas)
  . Fixed bug #67805 (SplFileObject setMaxLineLength). (Willian Gustavo Veiga).
  . Fixed bug #69970 (Use-after-free vulnerability in
    spl_recursive_it_move_forward_ex()). (Laruence)

- Sqlite3:
  . Fixed bug #69972 (Use-after-free vulnerability in
    sqlite3SafetyCheckSickOrOk()). (Laruence)

11 Jun 2015, PHP 5.6.10

- Core:
  . Fixed bug #66048 (temp. directory is cached during multiple requests).
    (Julien)
  . Fixed bug #69566 (Conditional jump or move depends on uninitialised value
    in extension trait). (jbboehr at gmail dot com)
  . Fixed bug #69599 (Strange generator+exception+variadic crash). (Nikita)
  . Fixed bug #69628 (complex GLOB_BRACE fails on Windows).
    (Christoph M. Becker)
  . Fixed POST data processing slowdown due to small input buffer size
    on Windows. (Jorge Oliveira, Anatol)
  . Fixed bug #69646 (OS command injection vulnerability in escapeshellarg).
    (CVE-2015-4642) (Anatol Belski)

contatos.php
<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>Website by Orlando Vieira</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">SMIR</h1>
      <h2 class="project-tagline">Servidor de Monitorização e Intervenção Remota</h2>
      <a href="index.html" class="btn">SMIR PI</a>
      <a href="hardware.php" class="btn">Hardware</a>
      <a href="dados.php" class="btn">Dados</a>
      <a href="graficos.php" class="btn">Gráficos</a></li>
      <a href="contatos.php" class="btn">Contatos</a></li>
  </section>
    <section class="main-content">

<clock>
<iframe src="http://free.timeanddate.com/clock/i4telzfn/n320/tlpt5/fn16/fs17/fcfff/tct/pct/tt0/tw0/tm1/ts1" frameborder="0" width="223" height="23" allowTransparency="true"></iframe>
</clock>

<body>
<br><h2> Projecto</h2>
SMIR: Servidor de Monitorização e Intervenção Remoto
<br><h2>Aluno</h2>
<p>Nome: Orlando Vieira
<p>Número: 1101421
<p>E-mail: 1101421@isep.ipp.pt

<h2>Orientador</h2>
<p>Nome: Armando Herculano
<p>E-mail:ahlf@isep.ipp.pt
</body>
</html>


hardware.php
<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>Website by Orlando Vieira</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">SMIR</h1>
      <h2 class="project-tagline">Servidor de Monitorização e Intervenção Remota</h2>
      <a href="index.html" class="btn">SMIR PI</a>
      <a href="hardware.php" class="btn">Hardware</a>
      <a href="dados.php" class="btn">Dados</a>
      <a href="graficos.php" class="btn">Gráficos</a></li>
      <a href="contatos.php" class="btn">Contatos</a></li>
  </section>
    <section class="main-content">

<clock>
<iframe src="http://free.timeanddate.com/clock/i4telzfn/n320/tlpt5/fn16/fs17/fcfff/tct/pct/tt0/tw0/tm1/ts1" frameborder="0" width="223" height="23" allowTransparency="true"></iframe>
</clock>

<body>
<h2> Hardware utilizado</h2>

<ul.a style="text-align=left">
  <li> Raspberry Pi 2</li><br>
  <li> Relógio de tempo real (DS1307) com bateria de backup através de I²C</li><br>
  <li> Sensor de Temperatura e Humidade DHT11</li><br>
  <li> Conversor Analógico/Digital MCP3008 - 8-Channel 10-Bit ADC com Interface SPI + multiplexer</li><br>
  <li> Potenciometro</li><br>
  <li> Painel Fotovoltaico</li><br>
  <li> Fios</li>
</ul.a>


Cenas: <code>gh-pages</code>
</body>
</html>


php_checkbox.php


<!DOCTYPE html>
<html>
<head>
	<title>PHP: Get Values of Multiple Checked Checkboxes</title>
	<link rel="stylesheet" href="css/php_checkbox.css" />
</head>
<body>
   <div class="container">
   <div class="main">
   	<h2>PHP: Get Values of Multiple Checked Checkboxes</h2>
	<form action="php_checkbox.php" method="post">
		<label class="heading">Select Your Technical Exposure:</label>
		<input type="checkbox" name="check_list[]" value="C/C++"><label>C/C++</label>
		<input type="checkbox" name="check_list[]" value="Java"><label>Java</label>
		<input type="checkbox" name="check_list[]" value="PHP"><label>PHP</label>
		<input type="checkbox" name="check_list[]" value="HTML/CSS"><label>HTML/CSS</label>
		<input type="checkbox" name="check_list[]" value="UNIX/LINUX"><label>UNIX/LINUX</label>
		<input type="submit" name="submit" Value="Submit"/>
<!----- Including PHP Script ----->
<?php include 'ascheckbox_value.php';?>
</form>
</div>
</div>
</body>
</html>




graficos (cópia).php
<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>Website by Orlando Vieira</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">SMIR</h1>
      <h2 class="project-tagline">Servidor de Monitorização e Intervenção Remota</h2>
      <a href="index.html" class="btn">Início</a>
      <a href="hardware.php" class="btn">Hardware</a>
      <a href="dados.php" class="btn">Dados</a>
      <a href="graficos.php" class="btn">Gráficos</a></li>
      <a href="contatos.php" class="btn">Contatos</a></li>
  </section>
    <section class="main-content">
<p>
<link rel="stylesheet" href="css/php_checkbox.css" />
</head>
   <div class="container">
      <div class="main">
	<form action="graficos.php" method="post">
		<label class="heading">Escolha o(s) gráfico(s) para visualizar:</label><br>
		<input type="checkbox" name="check_list[]" value="Temperatura"><label>Temperatura  </label>
		<input type="checkbox" name="check_list[]" value="Humidade"><label>Humidade  </label>
		<input type="checkbox" name="check_list[]" value="Painel"><label>Painel  </label>
		<input type="checkbox" name="check_list[]" value="Potenciometro"><label>Bateria  </label>
		<input type="submit" name="submit" Value="Submit"/>
</p>
<!----- Including PHP Script ----->
<?php 
	$x=0;
	include 'checkbox_value.php';
?>
	</form>
      </div>
   </div>
</head>

<!-------------------Gráfico do painel-------------------------->
<!--------Recebe dados--------->
<body>
<?php
if($x==1){
class MyDB extends SQLite3
{
    function __construct()
    {
        $this->open('test.db');//inicializar a base de dados
    }
}

$matrix=array(array());
$db = new MyDB();
$result = $db->query('SELECT * FROM DADOS');
$i=0;
   if($checked_count==1){
	while ($row = $result->fetchArray()) {
		$vetor[]=$row['Hora'];
		$vetor2[]=$row[$column1];
	}
   }
   else		
   if($checked_count==2){
   	while ($row = $result->fetchArray()) {
		$vetor[]=$row['Hora'];
		$vetor2[]=$row[$column1];
		$vetor3[]=$row[$column2];
	}	
   }
   else
   if($checked_count==3){
	while ($row = $result->fetchArray()) {
		$vetor[]=$row['Hora'];
		$vetor2[]=$row[$column1];
		$vetor3[]=$row[$column2];
		$vetor4[]=$row[$column3];
	}		
   }
   else
   if($checked_count==4){
	while ($row = $result->fetchArray()) {
		$vetor[]=$row['Hora'];
		$vetor2[]=$row[$column1];
		$vetor3[]=$row[$column2];
		$vetor4[]=$row[$column3];
		$vetor5[]=$row[$column4];
	}
   }
   $linhas=count($vetor);


   $matrix=array(array());

   for($i=0; $i<$linhas;$i++){	
	for($j=0; $j<$a+1;$j++){
		if($j==0){
			$matrix[$i][$j]=$vetor[$i] ;}
		if($j==1){
			$matrix[$i][$j]=$vetor2[$i];}
		if($j==2){
			$matrix[$i][$j]=$vetor3[$i];}
		if($j==3){
			$matrix[$i][$j]=$vetor4[$i];}
		if($j==4){
			$matrix[$i][$j]=$vetor5[$i];}
	}	
    }
}
?>

<!-----------Desenha gráfico------->
<script type="text/javascript" src="https://www.google.com/jsapi"></script>
  <script type="text/javascript">
    google.load('visualization', '1.1', {packages: ['line']});
    google.setOnLoadCallback(drawChart);

    function drawChart() {

      var data = new google.visualization.DataTable();

<?php	if($checked_count==1){?>
            data.addColumn('number', 'Hora');
            data.addColumn('number', '<?php echo $column1;?>');
<?php	}
   	else		
   	if($checked_count==2){?>
	    data.addColumn('timeofday', 'Hora');
            data.addColumn('number', '<?php echo $column1;?>');
            data.addColumn('number', '<?php echo $column2;?>');
<?php	}
  	else
   	if($checked_count==3){?>
	       data.addColumn('number', 'Hora');
            data.addColumn('number', '<?php echo $column1;?>');
            data.addColumn('number', '<?php echo $column2;?>');
            data.addColumn('number', '<?php echo $column3;?>');
<?php	}
   	else
   	if($checked_count==4){?>
	    data.addColumn('number', 'Hora');
            data.addColumn('number', '<?php echo $column1;?>');
            data.addColumn('number', '<?php echo $column2;?>');
            data.addColumn('number', '<?php echo $column3;?>');
            data.addColumn('number', '<?php echo $column4;?>');
<?php	}?>
       

        var array=[];
       <?php
                $val=count($matrix);
                $val2=$a+1;
                for($i=0;$i<$val;$i++)
                { ?>
                        array[<?php Print($i);?>]=[];
                        <?php
                        for($j=0;$j<$val2;$j++)
                        {  ?>
                                array[<?php Print($i);?>][<?php Print($j); ?>]=<?php Print($matrix[$i][$j]);?>;
                        <?php
                        }
                }
        ?>

<?php	if($checked_count==1){?>
            data.addRows([ <?php for($j=0;$j<$val;$j++){?>
               		[array[<?php Print($j);?>][0], 
			 array[<?php Print($j);?>][1]
			],
		<?php } ?>
	]);
<?php	}
   	else		
   	if($checked_count==2){?>
	    data.addRows([ <?php for($j=0;$j<$val;$j++){?>
               		[array[<?php Print($j);?>][0], 
			 array[<?php Print($j);?>][1], 
			 array[<?php Print($j);?>][2]
			],
		<?php } ?>
	]);
<?php	}
  	else
   	if($checked_count==3){?>
	      data.addRows([ <?php for($j=0;$j<$val;$j++){?>
               		[array[<?php Print($j);?>][0], 
			 array[<?php Print($j);?>][1], 
			 array[<?php Print($j);?>][2], 
			 array[<?php Print($j);?>][3]
			],
		<?php } ?>
	]);
<?php	}
   	else
   	if($checked_count==4){?>
		data.addRows([ <?php for($j=0;$j<$val;$j++){?>
               		[array[<?php Print($j);?>][0], 
			 array[<?php Print($j);?>][1], 
			 array[<?php Print($j);?>][2], 
			 array[<?php Print($j);?>][3], 
			 array[<?php Print($j);?>][4]
			],
			<?php } ?>
		]);
<?php	}?>
       
      var options = {
        chart: {
          title: 'Gráficos dos valores obtidos pelo Raspberry Pi'
 //         subtitle: 'in millions of dollars (USD)'
        },
	//backgroundColor: '#BDBDBD',
	legend: { position: 'top', alignment: 'start' },
        width: 1023,
        height: 500,
	hAxis:{format: '##,###'},
      };
      var chart = new google.charts.Line(document.getElementById('linechart_material'));
      chart.draw(data, google.charts.Line.convertOptions(options));
    }
</script>

<?php
if($x==1){
	echo 'Gráfico';
	echo '<div id="linechart_material"></div>';
}?>
</body>
</html>


graficos.php
<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>Website by Orlando Vieira</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">

			
    <section class="page-header">
      <h1 class="project-name">SMIR</h1>
      <h2 class="project-tagline">Servidor de Monitorização e Intervenção Remota</h2>
      <a href="index.html" class="btn">Início</a>
      <a href="hardware.php" class="btn">Hardware</a>
      <a href="dados.php" class="btn">Dados</a>
      <a href="graficos.php" class="btn">Gráficos</a></li>
      <a href="contatos.php" class="btn">Contatos</a></li>
      
	<form action="graficos.php" method="post">
		<label class="heading">Escolha o(s) gráfico(s) para visualizar:</label><br>
		<input type="checkbox" name="check_list[]" value="Temperatura"><label>Temperatura  </label>
		<input type="checkbox" name="check_list[]" value="Humidade"><label>Humidade  </label>
		<input type="checkbox" name="check_list[]" value="Painel"><label>Painel  </label>
		<input type="checkbox" name="check_list[]" value="Potenciometro"><label>Bateria  </label>
		<input type="submit" name="submit" Value="Submit"/>
				<!----- Including PHP Script ----->
	<?php 
		$x=0;
		include 'checkbox_value.php';
	?>
	</form>
   </section>
</head>

<!-------------------Gráfico do painel-------------------------->
<!--------Recebe dados--------->
<body>
<?php
if($x==1){
class MyDB extends SQLite3
{
    function __construct()
    {
        $this->open('bd/lele.db');//inicializar a base de dados
    }
}

$matrix=array(array());
$db = new MyDB();
$result = $db->query('SELECT * FROM DADOS');
$i=0;
   if($checked_count==1){
	while ($row = $result->fetchArray()) {
		$vetor[]=$row['Hora'];
		$vetor2[]=$row[$column1];
	}
   }
   else		
   if($checked_count==2){
   	while ($row = $result->fetchArray()) {
		$vetor[]=$row['Hora'];
		$vetor2[]=$row[$column1];
		$vetor3[]=$row[$column2];
	}	
   }
   else
   if($checked_count==3){
	while ($row = $result->fetchArray()) {
		$vetor[]=$row['Hora'];
		$vetor2[]=$row[$column1];
		$vetor3[]=$row[$column2];
		$vetor4[]=$row[$column3];
	}		
   }
   else
   if($checked_count==4){
	while ($row = $result->fetchArray()) {
		$vetor[]=$row['Hora'];
		$vetor2[]=$row[$column1];
		$vetor3[]=$row[$column2];
		$vetor4[]=$row[$column3];
		$vetor5[]=$row[$column4];
	}
   }
   $linhas=count($vetor);


   $matrix=array(array());

   for($i=0; $i<$linhas;$i++){	
	for($j=0; $j<$a+1;$j++){
		if($j==0){
			$matrix[$i][$j]=$vetor[$i] ;}
		if($j==1){
			$matrix[$i][$j]=$vetor2[$i];}
		if($j==2){
			$matrix[$i][$j]=$vetor3[$i];}
		if($j==3){
			$matrix[$i][$j]=$vetor4[$i];}
		if($j==4){
			$matrix[$i][$j]=$vetor5[$i];}
	}	
    }
}
?>

<!-----------Desenha gráfico------->
<script type="text/javascript" src="https://www.google.com/jsapi"></script>
  <script type="text/javascript">
    google.load('visualization', '1.1', {packages: ['line']});
    google.setOnLoadCallback(drawChart);

    function drawChart() {

      var data = new google.visualization.DataTable();

<?php	if($checked_count==1){?>
            data.addColumn('number', 'Hora');
            data.addColumn('number', '<?php echo $column1;?>');
<?php	}
   	else		
   	if($checked_count==2){?>
	    data.addColumn('timeofday', 'Hora');
            data.addColumn('number', '<?php echo $column1;?>');
            data.addColumn('number', '<?php echo $column2;?>');
<?php	}
  	else
   	if($checked_count==3){?>
	       data.addColumn('number', 'Hora');
            data.addColumn('number', '<?php echo $column1;?>');
            data.addColumn('number', '<?php echo $column2;?>');
            data.addColumn('number', '<?php echo $column3;?>');
<?php	}
   	else
   	if($checked_count==4){?>
	    data.addColumn('number', 'Hora');
            data.addColumn('number', '<?php echo $column1;?>');
            data.addColumn('number', '<?php echo $column2;?>');
            data.addColumn('number', '<?php echo $column3;?>');
            data.addColumn('number', '<?php echo $column4;?>');
<?php	}?>
       

        var array=[];
       <?php
                $val=count($matrix);
                $val2=$a+1;
                for($i=0;$i<$val;$i++)
                { ?>
                        array[<?php Print($i);?>]=[];
                        <?php
                        for($j=0;$j<$val2;$j++)
                        {  ?>
                                array[<?php Print($i);?>][<?php Print($j); ?>]=<?php Print($matrix[$i][$j]);?>;
                        <?php
                        }
                }
        ?>

<?php	if($checked_count==1){?>
            data.addRows([ <?php for($j=0;$j<$val;$j++){?>
               		[array[<?php Print($j);?>][0], 
			 array[<?php Print($j);?>][1]
			],
		<?php } ?>
	]);
<?php	}
   	else		
   	if($checked_count==2){?>
	    data.addRows([ <?php for($j=0;$j<$val;$j++){?>
               		[array[<?php Print($j);?>][0], 
			 array[<?php Print($j);?>][1], 
			 array[<?php Print($j);?>][2]
			],
		<?php } ?>
	]);
<?php	}
  	else
   	if($checked_count==3){?>
	      data.addRows([ <?php for($j=0;$j<$val;$j++){?>
               		[array[<?php Print($j);?>][0], 
			 array[<?php Print($j);?>][1], 
			 array[<?php Print($j);?>][2], 
			 array[<?php Print($j);?>][3]
			],
		<?php } ?>
	]);
<?php	}
   	else
   	if($checked_count==4){?>
		data.addRows([ <?php for($j=0;$j<$val;$j++){?>
               		[array[<?php Print($j);?>][0], 
			 array[<?php Print($j);?>][1], 
			 array[<?php Print($j);?>][2], 
			 array[<?php Print($j);?>][3], 
			 array[<?php Print($j);?>][4]
			],
			<?php } ?>
		]);
<?php	}?>
       
      var options = {
        chart: {
          title: 'Gráficos dos valores obtidos pelo Raspberry Pi'
 //         subtitle: 'in millions of dollars (USD)'
        },
	//backgroundColor: '#BDBDBD',
	legend: { position: 'top', alignment: 'start' },
        width: 1023,
        height: 500,
	hAxis:{format: '##,###'},
      };
      var chart = new google.charts.Line(document.getElementById('linechart_material'));
      chart.draw(data, google.charts.Line.convertOptions(options));
    }
</script>

<?php

  if($x==1){	
	//if($checked_count==1)
	//{
//		echo "".$checked_count." Opção selecionada: <br/>";
//	}		
//	if($checked_count>1)
//		echo "".$checked_count." Opções selecionadas: <br/>";
		// Loop to store and display v alues of individual checked checkbox.
//	echo '<br>';
	echo '<div id="linechart_material" align="center"></div>';
  }
  if($x==0){?>
		  
   <div class="container">
	    <div class="alert alert-warning">
	      <a href="#" class="close" data-dismiss="alert" aria-label="close">&times;</a>
	      <strong>Aviso!</strong> Selecione pelo menos uma variável para poder visualizar o gráfico!
	    </div>
	  </div>
<?php }
?>
</body>
</html>
